nens_auth_client/backends.py

```diff
@@ -112,3 +112,47 @@ def authenticate(self, request, claims):
         create_remote_user(user, claims)
 
         return user if self.user_can_authenticate(user) else None
+
+
+class AcceptNensBackend(ModelBackend):
+    def authenticate(self, request, claims):
+        """Backend for auto-accepting users that have a N&S azure AD account.
+
+        The behaviour looks a bit like the SSOMigrationBackend above, but with
+        two key differences:
+
+        - N&S users don't need an existing account, they're accepted right
+          away and a User is created if missing.
+
+        - We only deal with users recognized as being really from N&S.
+
+        Args:
+          request: the current request
+          claims (dict): the verified payload of the ID or Access token
+
+        Returns:
+          user or None
+
+        """
+        username = _nens_user_extract_username(claims)
+        if username is None:
+            return
+
+        email = claims.get("email")
+
+        try:
+            user, created = UserModel.objects.get_or_create(
```

Comment:
Werkt een `get_or_create` met filters zoals deze? Is django slim genoeg om de `__iexact` filters eruit te slopen als hij een create gaat doen? Zou ik niet kunnen zeggen uit mijn hoofd.

Misschien goed om een keertje te testen (het zit in in de unittests zie ik)

Reply:
De unittests zijn één groot mock-festijn op dat punt, dus alleen de mock van get_or_create wordt getest.

Ik zal eens wat handmatig gaan testen.

Reply:
Volgens https://docs.djangoproject.com/en/4.0/ref/models/querysets/#get-or-create wordt alles met een dubbele underscore wel gebruikt voor de "get", maar niet voor de "create". Aha! Dat ga ik fixen.